### PR TITLE
test: remove infinite timeout

### PR DESCRIPTION
**Pull request by NoamGaash**

### CODE REVIEW

File: src/test_pages/BasePage.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export abstract class BasePage {
   protected async clickOnElement(element: Locator, timeout?: number) {
     // TODO: make sure that element.toString() doesn't make [object Object]
     await test.step(`Click on ${element.toString()}`, async () => {
-      await element.click({ timeout: timeout || 0 })
+      await element.click({ timeout: timeout })
```
Comment (NoamGaash) [inline src/test_pages/BasePage.ts:15]:
```suggestion
      await element.click({ timeout })
```

File: src/test_pages/BasePage.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export abstract class BasePage {
   protected async clickOnElement(element: Locator, timeout?: number) {
     // TODO: make sure that element.toString() doesn't make [object Object]
     await test.step(`Click on ${element.toString()}`, async () => {
-      await element.click({ timeout: timeout || 0 })
+      await element.click({ timeout })
```
Comment (R894) [inline src/test_pages/BasePage.ts:15]: Looks great!
I took a look at Playwright documentation noticed a small thing,  ```timeout``` will default to ```0``` when undefined, 
If we want to change this behavior we can also modify the ```playwright.config.ts``` -
 ``` { use:  { ..., actionTimeout: 10000 }}```

**Review by R894 — CHANGES_REQUESTED** [review]
Looks great!
I took a look at Playwright documentation noticed a small thing,  ```timeout``` will default to ```0``` when undefined, 
If we want to change this behavior we can also modify the ```playwright.config.ts``` -
 ``` { use:  { ..., actionTimeout: 10000 }}```

**Comment by NoamGaash** [comment]
AFAIK playwright has a default 5 seconds timeout
https://playwright.dev/docs/test-timeouts

File: src/test_pages/BasePage.ts (hunk 2 above)
Comment (R894) [inline src/test_pages/BasePage.ts:15]:
```suggestion
      expect(element).not.toBeNull()
      await element.click({ timeout })
```
I think the problem is that element.click() bypasses the ```expect``` timeout of 5 seconds, leaving it to hang until the ```test``` timeout

**Review by R894 — APPROVED** [review]

File: src/test_pages/BasePage.ts (hunk 2 above)
Comment (NoamGaash) [reply src/test_pages/BasePage.ts:15]: I'm pretty sure that `actionTimeout` is `5000ms` by default.
`element` is a Locator object, so it won't be null even if the locator won't resolve to any dom element. You probably mean you want to do something like `expect(locator).toBeVisible()` or `expect(locator).toBeEnabled()`, but the [auto-waiting](https://playwright.dev/docs/actionability) magic will do it for us so it's not necessary

**Pull request merged**